tiledb/cloud/pool_manager_wrapper.py

```diff
@@ -1,17 +1,41 @@
+import urllib
+
+
 class _PoolManagerWrapper:
     """
     Wrapper around urllib3.PoolManager that implements request() which
     is the only method used by the autogenerated API code
 
     It can be used to replace tiledb.cloud.rest_api.rest.RESTClientObject.pool_manager
-    at runtime to work around https://github.com/urllib3/urllib3/issues/2475
+    at runtime in order to cache HTTP redirects and work around
+    https://github.com/urllib3/urllib3/issues/2475
     """
 
     def __init__(self, pool_manager):
         self._pool = pool_manager
+        self._redirect_cache = {}
```

Comment:
with an added `from typing import Dict` above, add:

```suggestion
        self._redirect_cache: Dict[str, str] = {}
```

Also, can you add a comment explaining exactly what the semantics of the keys and values are? It looks like it's `{url without query string: domain}`.

Reply:
fixed